### PR TITLE
Improve the design of various components

### PR DESCRIPTION
**Pull request by double-beep**
- Use `GM_setValue`, `GM_getValue` and `GM_deleteValue` if the user uses a userscript manager, otherwise store config in `localStorate`.
- Append the config modal to the body. Convert margin to padding to fix an issue where part of the outline of the switch was hidden.
- Use [Stacks' user cards](https://stackoverflow.design/product/components/user-cards/) for the user cards of suggestors, editors, and askers/answerers. [before](https://user-images.githubusercontent.com/38133098/123046558-ead94f80-d404-11eb-957a-1324f1c75120.png)/[after](https://user-images.githubusercontent.com/38133098/123045200-499dc980-d403-11eb-9ed4-6ba8f01abfbc.png)
- Use modern `async/await` syntax on `fetch`es instead of `.then()`/`.catch()`.
- Show the [user cards not found](https://i.stack.imgur.com/EL5NH.png) error message when the edit has been suggested on a deleted answer.

### CODE REVIEW

File: StackReviewSuggestedEditsRework.user.js [hunk 1]
```diff
@@ -2185,22 +2185,14 @@
...
-                settingsLink.dataset.action = show;
-                settingsItem.dataset.controller = modal;
-
-                // The GUI needs to load first..
-                sleep(50).then(() => {
-                    settingsLink.click();
-                    // Stacks.showModal(settingsItem);
-                })
-            }, { once: true });
+                setTimeout(() => Stacks.showModal(modalElement), 0);
```
Comment (Oaphi) [inline StackReviewSuggestedEditsRework.user.js]: btw, can somebody explain what the heck are we doing there? I mean I get this is a timeout scheduled for the "immediate" execution, but maybe we have a better way to determine if everything has loaded?

File: StackReviewSuggestedEditsRework.user.js [hunk 2]
```diff
@@ -711,66 +711,45 @@
...
-                                      ];
-
-                     userCardsContainerAll.appendChild(missingCards(messages));
-              });
+            try {
+                const userCardRequest = await fetch(postlink);
+                const userCardResponse = await userCardRequest.text();
+                // https://developer.mozilla.org/en-US/docs/Web/API/DOMParser
+                // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals
+                if (userCardRequest.status === 200) {
```
Comment (Oaphi) [inline StackReviewSuggestedEditsRework.user.js]: not sure I like the `if...else if...else` structure over maps. @Scratle?

File: StackReviewSuggestedEditsRework.user.js [hunk 3]
```diff
@@ -669,6 +669,42 @@
         }
     }
 
+    // -------    createUserCard    --------------------
+    // official Stacks documentation: https://stackoverflow.design/product/components/user-cards/
+    function createUserCard(isUserOwner, actionText, actionISO, profileUrl, profileImage, username, reputation, badges) {
```
Comment (Oaphi) [inline StackReviewSuggestedEditsRework.user.js]: maybe switch the public contract to accept a single configuration object? It's too many positional parameters to not do that, case in point:

```javascript
createUserCard(false, proposedText, proposedISO, null, null, null, null, {});
```

File: StackReviewSuggestedEditsRework.user.js [hunk 4]
```diff
@@ -793,7 +793,9 @@
             // -------    stacksifyUserCards    --------------------
             function stacksifyUserCards(original) {
                 const userActionTime = original.querySelector(config.selectors.userCards.actionTime);
-                const actionText = userActionTime.innerText; // e.g. asked 4 hours ago, edited Sep 9 '19 at 10:25
+                // e.g. asked 4 hours ago, edited Sep 9 '19 at 10:25
+                // if it's the edited usercard, then the element needs to be an anchor pointing to the revision history
+                const actionText = userActionTime.innerText.includes('edited') ? userActionTime.outerHTML : userActionTime.innerText;
```
Comment (Oaphi) [inline StackReviewSuggestedEditsRework.user.js]: How about this (should be a little bit cleaner):

```javascript
const { innerText, outerHTML } = userActionTime;
const actionText = innerText.includes('edited') ? outerHTML : innerText;
```

**Review by Oaphi — APPROVED** [review]
I left some comments but they aren't blocking - LGTM

File: StackReviewSuggestedEditsRework.user.js (hunk 2 above)
Comment (double-beep) [reply StackReviewSuggestedEditsRework.user.js]: @Oaphi this was updated in https://github.com/Scratle/fiddlings/pull/2/commits/70e3e7a68df11f33b74ba48d84ea75d519ab3aef to look better, does it still need a map?

Comment (Oaphi) [reply StackReviewSuggestedEditsRework.user.js]: Methinks yes, I hold a particular dislike for the `else if` construct - but it's a non-blocking thing

File: StackReviewSuggestedEditsRework.user.js (hunk 1 above)
Comment (double-beep) [reply StackReviewSuggestedEditsRework.user.js]: No idea why `Stacks.showModal()` doesn't work, but `StackExchange.helpers.showModal();` for some reason does. Do you think we could use the latter or load the modal for the first time on hover?

Comment (Oaphi) [reply StackReviewSuggestedEditsRework.user.js]: > No idea why `Stacks.showModal()` doesn't work, but `StackExchange.helpers.showModal();` for some reason does. Do you think we could use the latter or load the modal for the first time on hover?

mm, something tells me there is a bug in `showModal` (sorry for a bit off-topic thing, just noticed I was looking at the `toggleModal` method). Take a look at [this line](https://github.com/StackExchange/Stacks/blob/8b49250a97b3da3f6be31b6de0f843697388c97b/lib/ts/controllers/s-modal.ts#L320). The JSDoc comment says it "toggles the modal if left undefined" but due to how they check for truthiness, the modal always gets hidden. 

Maybe we should switch to `StackExchange` implementation for now

Comment (double-beep) [reply StackReviewSuggestedEditsRework.user.js]: Hmm... With the `StackExchange` method, if I click the gear icon, then `x` or Cancel and then again the gear icon, the modal opens for a moment and then it's hidden/removed again.

File: StackReviewSuggestedEditsRework.user.js [hunk 5]
```diff
@@ -237,11 +237,11 @@
...
-        if (choices) {
-            moveToFilterLine(choices, true);
-            choices.firstElementChild.style.padding = "10px";  // the larger padding on one takes precedence
-            choices.id = diffChoices;
-        }
+        if (!choices) return;
+
+        moveToFilterLine(choices, true);
+        choices.id = diffChoices;
+        [...choices.children].forEach((button) => button.classList.remove("s-btn__xs"));
```
Comment (Scratle) [inline StackReviewSuggestedEditsRework.user.js:244]: I tried this with buttons and not moving the progress.  ["s-btn__xs" and padding](https://i.stack.imgur.com/Yjpyt.png) seems do-able to me. [No "s-btn__xs" even without padding](https://i.stack.imgur.com/OJZQ5.png) is a bit too cramped up.

**Review by Scratle — COMMENTED** [review]
I tried this with buttons and not moving the progress.  ["s-btn__xs" and padding](https://i.stack.imgur.com/Yjpyt.png) seems do-able to me. [No "s-btn__xs" even without padding](https://i.stack.imgur.com/OJZQ5.png) is a bit too cramped up. This was supposed to be a comment on [this commit](https://github.com/Scratle/fiddlings/pull/2/commits/6327f8fd44531b92ff66798fbfa6c09498665be5), but I can't figure out how to do that.

**Review by Scratle — COMMENTED** [review]
(deleted) Sorry about this, but I'm [not allowed to delete a "Review changed" comment](https://i.stack.imgur.com/aoLYL.png)

**Review by Oaphi — COMMENTED** [review]
this is a test review comment

**Review by Oaphi — COMMENTED** [review]
We can ditch the `[...el.querySelectorAll("...")].forEach` and just do: `el.querySelectorAll("...").forEach` as the method exists on the collection returned

**Review by Oaphi — COMMENTED** [review]
btw, I suppose we can move the `"pe-none"` class to config object

File: StackReviewSuggestedEditsRework.user.js [hunk 6]
```diff
@@ -3239,10 +3239,8 @@
 
             radios.forEach((radio) => {
```
Comment (Oaphi) [inline StackReviewSuggestedEditsRework.user.js:3328]: We should probably optimize here:

```javascript
radios.forEach(({ style }) => {
   style.borderLeft = radioWithBorders ? `${radioSeperatorColour} solid ${radioSeperatorSize}px` : "none";
});
```

**Review by Oaphi — COMMENTED** [review]
We should probably wrap the `new Date().toISOString().replace("T", " ")` into a utility

**Pull request merged**